pkg/apis/networking/validation/validation.go

```diff
@@ -174,15 +182,47 @@ func ValidateIPBlock(ipb *networking.IPBlock, fldPath *field.Path) field.ErrorLi
 	return allErrs
 }
 
-// ValidateIngress tests if required fields in the Ingress are set.
-func ValidateIngress(ingress *networking.Ingress) field.ErrorList {
+// ValidateIngressName validates that the given name can be used as an Ingress
+// name.
+var ValidateIngressName = apimachineryvalidation.NameIsDNSSubdomain
+
+
+// Validation options for new beta to GA transitions for HTTP PathType
+type IngressValidationOptions struct {
+	requireRegexPath bool
+}
+
+// ValidateIngress validates Ingresses on create and update.
+func ValidateIngress(ingress *networking.Ingress, opts IngressValidationOptions, requestGV schema.GroupVersion) field.ErrorList {
```

Comment:
suggest making ValidateIngress private so no external callers accidentally call this instead of the Create or Update-specific validation